mscore/pagesettings.cpp

```diff
@@ -270,311 +325,430 @@ void PageSettings::orientationClicked()
-            }
-      Score* s = preview->score();
-      s->style().set(Sid::pageEvenLeftMargin, val);
-      s->style().set(Sid::pagePrintableWidth, s->styleD(Sid::pageWidth) - evenPageRightMargin->value() * f - val);
-      updatePreview(0);
+      lrMargins(val, false, false, evenRightMargin);
       }
-
+//// end margins ////
 //---------------------------------------------------------
-//   ermChanged
+//   spatiumChanged
 //---------------------------------------------------------
 
-void PageSettings::ermChanged(double val)
+void PageSettings::spatiumChanged(double val)
       {
-      if (mmUnit)
-            val /= INCH;
+      Score* score  = preview->score();
+      double oldVal = score->spatium();
+      double newVal;
+
+      QPageLayout::Unit unit = score->style().pageOdd().units();
+      if ((unit == QPageLayout::Millimeter && val == 1.764) ||
+          (unit == QPageLayout::Inch       && val == 0.069) ||
+          (unit == QPageLayout::Pica       && val == 0.417) ||
+          (unit == QPageLayout::Didot      && val == 4.692) ||
+          (unit == QPageLayout::Cicero     && val == 0.391))
+            newVal = SPATIUM20; // rounding messes up the default value
+      else
+            newVal = val * pageUnits[int(unit)].paintFactor();
 
-      if (twosided->isChecked()) {
-            oddPageLeftMargin->blockSignals(true);
-            oddPageLeftMargin->setValue(val * (mmUnit ? INCH : 1.0));
-            oddPageLeftMargin->blockSignals(false);
-            }
-      Score* s = preview->score();
-      s->style().set(Sid::pagePrintableWidth, s->styleD(Sid::pageWidth) - s->styleD(Sid::pageEvenLeftMargin) - val);
-      s->style().set(Sid::pageOddLeftMargin, val);
-      updatePreview(0);
+      score->setSpatium(newVal);
+      score->spatiumChanged(oldVal, newVal);
+      updatePreview();
       }
 
 //---------------------------------------------------------
-//   ebmChanged
+//   pageOffsetChanged
 //---------------------------------------------------------
 
-void PageSettings::ebmChanged(double val)
+void PageSettings::pageOffsetChanged(int val)
       {
-      if (mmUnit)
-            val /= INCH;
-      preview->score()->style().set(Sid::pageEvenBottomMargin, val);
-      updatePreview(1);
+      preview->score()->setPageNumberOffset(val-1);
+      updatePreview();
       }
 
 //---------------------------------------------------------
-//   spatiumChanged
+//   unitsChanged
 //---------------------------------------------------------
 
-void PageSettings::spatiumChanged(double val)
+void PageSettings::unitsChanged()
       {
-      val *= mmUnit ? DPMM : DPI;
-      double oldVal = preview->score()->spatium();
-      preview->score()->setSpatium(val);
-      preview->score()->spatiumChanged(oldVal, val);
-      updatePreview(0);
+      QPageLayout::Unit u = QPageLayout::Unit(unitsList->currentIndex());
+      MStyle& style = preview->score()->style();
+      style.pageOdd ().setUnits(u);
```

Comment:
Spaces

Reply:
I already fixed this but haven't yet pushed.  I've done a lot of space removing.